components/aws/sagemaker/tests/integration_tests/utils/kfp_client_utils.py

```diff
@@ -61,3 +69,30 @@ def compile_run_monitor_pipeline(
         pytest.fail(f"Test Failed: {pipeline_name}. Run-id: {run_id}")
 
     return run_id, status, workflow_json
+
+
+def compile_run_pipeline(
```

Comment:
there is scope to reduce some code duplication here,

how about we add status as a parameter to `compile_run_monitor_pipeline()` with default value as `succeeded` and change the wait_for_job_completion to accept it? maybe change the name to wait_for_job_status. Then you could call this method with `compile_run_monitor_pipeline(. . .,status='running', check=False)`

and rather than having a constant value of 10 second sleep, define it in test's yaml file as the timeout/waitTime. Wdyt?

When passing check=False, do print the workflow logs in the test before running assertions

Reply:
fair point, but again I assumed this method will be removed once we have a bot in place so this separation makes it easy to clean code later.

Reply:
at that time we just increase the timeout value and can use the default value for status

Reply:
`compile_run_monitor_pipeline` is still a valid name :)

Reply:
Done.